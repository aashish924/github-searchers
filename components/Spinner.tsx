import React from 'react';
import styled from 'styled-components';


const SpinnerContainer = styled.span`
  position: relative;
  display: inline-block;
  text-align: center;
  width: 120px;
  height: 24px;
  left: calc(30% - 120px);
  margin: 16px;
  overflow: hidden;
  line-height: 1.5em;
  font-size: 18px;
  font-weight: 400;

  &::after {
    display: inline-table;
    white-space: pre;
    content: "🕐\\A🕑\\A🕒\\A🕓\\A🕔\\A🕕\\A🕖\\A🕗\\A🕘\\A🕙\\A🕚\\A🕛";
    animation: spin12 3s steps(12) infinite;
    width: 1.5em;
  }

  @keyframes spin12 {
    to {
      transform: translateY(-18.0em);
    }
  }
`;

export const Spinner = ({ text }: { text?: string}) => {
    return <SpinnerContainer>{text}</SpinnerContainer>
};

export default Spinner;
