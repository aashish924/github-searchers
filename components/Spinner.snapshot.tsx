import * as React from 'react';
import renderer from 'react-test-renderer';

import Spinner from './Spinner'


describe('Spinner', () => {
  it('Spinner without text should render correctly', () => {
    const tree = renderer.create(<Spinner />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Spinner with text should render correctly', () => {
    const tree = renderer.create(<Spinner text='loading...' />).toJSON();
    expect(tree).toMatchSnapshot();
  });
})