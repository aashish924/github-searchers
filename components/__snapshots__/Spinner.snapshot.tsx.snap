// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Spinner with text should render correctly 1`] = `
<span
  className="sc-AxjAm bpLAqs"
>
  loading...
</span>
`;

exports[`Spinner Spinner without text should render correctly 1`] = `
<span
  className="sc-AxjAm bpLAqs"
/>
`;
